### PR TITLE
Add challenge tests for logsdb that utilize stored source

### PR DESCRIPTION
**Pull request by lkts**

### CODE REVIEW

File: x-pack/plugin/logsdb/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/qa/LogsDbVersusLogsDbReindexedIntoStandardModeChallengeRestIT.java [hunk 1]
```diff
@@ -52,27 +45,4 @@ public void baselineMappings(XContentBuilder builder) throws IOException {
     public void contenderMappings(XContentBuilder builder) throws IOException {
         dataGenerationHelper.standardMapping(builder);
     }
-
-    @Override
-    public Response indexContenderDocuments(CheckedSupplier<List<XContentBuilder>, IOException> documentsSupplier) throws IOException {
```
Comment (lkts) [inline x-pack/plugin/logsdb/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/qa/LogsDbVersusLogsDbReindexedIntoStandardModeChallengeRestIT.java:57]: This method exists in a base class, just a clean up.

File: x-pack/plugin/logsdb/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/qa/LogsDbVersusReindexedLogsDbChallengeRestIT.java [hunk 2]
```diff
@@ -52,27 +45,4 @@ public void baselineMappings(XContentBuilder builder) throws IOException {
     public void contenderMappings(XContentBuilder builder) throws IOException {
         dataGenerationHelper.logsDbMapping(builder);
     }
-
-    @Override
-    public Response indexContenderDocuments(CheckedSupplier<List<XContentBuilder>, IOException> documentsSupplier) throws IOException {
```
Comment (lkts) [inline x-pack/plugin/logsdb/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/qa/LogsDbVersusReindexedLogsDbChallengeRestIT.java:57]: This method exists in a base class, just a clean up.

File: x-pack/plugin/logsdb/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/qa/DataGenerationHelper.java [hunk 3]
```diff
@@ -89,6 +89,7 @@ void standardMapping(XContentBuilder builder) throws IOException {
     }
 
     void logsDbSettings(Settings.Builder builder) {
+        builder.put("index.mode", "logsdb");
```
Comment (lkts) [inline x-pack/plugin/logsdb/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/qa/DataGenerationHelper.java:92]: Noticed this oversight, it only impacted reindex tests.

**Review by kkrik-es — APPROVED** [review]

**Review by salvatore-campagna — APPROVED** [review]

File: x-pack/plugin/logsdb/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/qa/LogsDbVersusReindexedIntoStoredSourceChallengeRestIT.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        return "logs-apache-reindexed";
+    }
+
+    @Override
+    public void baselineSettings(Settings.Builder builder) {
+        dataGenerationHelper.logsDbSettings(builder);
+    }
+
+    @Override
+    public void contenderSettings(Settings.Builder builder) {
```
Comment (salvatore-campagna) [inline x-pack/plugin/logsdb/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/qa/LogsDbVersusReindexedIntoStoredSourceChallengeRestIT.java:34]: I am wondering if we should just create base classes like

```
SyntheticSourceAbstractChallengeRestTest
DisableSourceAbstractChallengeRestTest
StoredSourceAbstractChallengeRestTest
```

where the setting is applied and we can just inherit from them.

Comment (lkts) [reply x-pack/plugin/logsdb/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/qa/LogsDbVersusReindexedIntoStoredSourceChallengeRestIT.java:34]: The thing is that sometimes baseline has stored source and sometimes the contender.

**Comment by lkts** [comment]
@elasticmachine update branch

**Comment by lkts** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/115699)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

**Pull request merged**